src/vos/vos_io.c

```diff
@@ -880,9 +888,22 @@ akey_update(struct vos_io_context *ioc, uint32_t pm_ver, daos_handle_t ak_toh)
 		goto out;
 	} /* else: array */
 
-	for (i = 0; i < iod->iod_nr; i++) {
+	for (i = 0; i < iod->iod_nr; i++, ioc->ic_umoffs_at++) {
```

Comment:
Is this not incremented elsewhere?   Or was this a bug?

Reply:
It is not incremented elsewhere.
Originally, we will not skip any IOD. But in the daos_racer test, the client offers some invalid IOD parameters that contains zero "rx_nr", then we have to skip those IODs. Under such case, related SCM reserve slots also need to be skipped to match the IOD's skip.

Reply:
It is updated in iod_update_umoff so this isn't quite right.  We only want to increment it when rx_nr == 0.

Reply:
Based on the functional test results, I suspect we may not be testing any multi-recx updates

Reply:
We need to increment the ic_umoffs_at for every IOD, not just for "rx_nr==0" case, because the "ioc->ic_umoffs" slots and the IODs should be paired. Otherwise, we will hit assertion() at the following line 898, that has been verified by daos_racer.

Reply:
I think it was a bug. Each reserve will add a umoff in the ic_umoffs array (even for record punch, we added a NULL in the array), however, the cursor of this array is only updated when updating SV (see iod_update_umoff() in akey_update_single(), EV updating doesn't require this umoff). I think we didn't hit error because all existing test only do either SV or EV updating in single PRC, no mixed cases.

Could you call the iod_update_umoff() in this loop as well? Sorry for introducing this crap code. :)

Reply:
>  I think we didn't hit error because all existing test only do either SV or EV updating in single PRC, no mixed cases.

Yes, it is true. But the does_racer test (not enabled yet) mixes the two types, then hit trouble. I will update the patch with iod_update_umoff() called to unify the style.